airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/integration/test_report_based_streams.py

```diff
@@ -0,0 +1,423 @@
+    @HttpMocker()
+    def test_given_report_access_forbidden_when_read_then_no_records_and_error_logged(
+        self, stream_name: str, data_format: str, http_mocker: HttpMocker, caplog: Any
+    ) -> None:
+        mock_auth(http_mocker)
+
+        http_mocker.post(_create_report_request(stream_name).build(), response_with_status(status_code=HTTPStatus.FORBIDDEN))
+
+        output = self._read(stream_name, config())
+        message_on_access_forbidden = (
+            "This is most likely due to insufficient permissions on the credentials in use. "
+            "Try to grant required permissions/scopes or re-authenticate."
+        )
+        assert_message_in_output(message_on_access_forbidden, caplog)
+        assert len(output.records) == 0
+
+    @pytest.mark.parametrize(("stream_name", "data_format"), STREAMS)
+    @HttpMocker()
+    def test_given_report_status_cancelled_when_read_then_no_records_and_error_logged(
+        self, stream_name: str, data_format: str, http_mocker: HttpMocker, caplog: Any
+    ) -> None:
+        mock_auth(http_mocker)
+
+        http_mocker.post(_create_report_request(stream_name).build(), _create_report_response())
+        http_mocker.get(
+            _check_report_status_request(_REPORT_ID).build(),
+            _check_report_status_response(stream_name, processing_status=ReportProcessingStatus.CANCELLED),
+        )
+
+        message_on_report_cancelled = f"The report for stream '{stream_name}' was cancelled or there is no data to return"
+
+        output = self._read(stream_name, config())
+        assert_message_in_output(message_on_report_cancelled, caplog)
+        assert len(output.records) == 0
+
+    @pytest.mark.parametrize(("stream_name", "data_format"), STREAMS)
+    @HttpMocker()
+    def test_given_report_status_fatal_when_read_then_exception_raised(
+        self, stream_name: str, data_format: str, http_mocker: HttpMocker
+    ) -> None:
+        mock_auth(http_mocker)
+
+        http_mocker.post(_create_report_request(stream_name).build(), _create_report_response())
+        http_mocker.get(
+            _check_report_status_request(_REPORT_ID).build(),
+            _check_report_status_response(stream_name, processing_status=ReportProcessingStatus.FATAL),
+        )
+
+        output = self._read(stream_name, config(), expecting_exception=True)
+        assert output.errors[-1].trace.error.failure_type == FailureType.system_error
+        assert output.errors[-1].trace.error.message == f"The report for stream '{stream_name}' was not created - skip reading"
+
+    @pytest.mark.parametrize(
+        ("stream_name", "date_field", "expected_date_value"),
+        (
+            ("GET_SELLER_FEEDBACK_DATA", "date", "2020-10-20"),
+            ("GET_LEDGER_DETAIL_VIEW_DATA", "Date", "2021-11-21"),
+            ("GET_LEDGER_SUMMARY_VIEW_DATA", "Date", "2022-12-22"),
+        ),
+    )
+    @HttpMocker()
+    def test_given_report_with_incorrect_date_format_when_read_then_formatted(
```

Comment:
This will be possible once the framework supports records builder for `.csv` templates format. Can we make some refactor after the framework updated or this test change should it be made before merge?